src/test/java/com/google/cloud/tools/managedcloudsdk/internal/download/StandardDownloaderTest.java

```diff
@@ -0,0 +1,179 @@
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.internal.download;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.nio.charset.Charset;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.NotDirectoryException;
+import java.nio.file.Path;
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+public class StandardDownloaderTest {
+
+  @Rule public TemporaryFolder tmp = new TemporaryFolder();
+
+  @Test
+  public void testValidateAndCreateFiles_pass() throws IOException {
+    Path destination = tmp.getRoot().toPath().resolve("destinationFile");
+
+    StandardDownloader downloader = new StandardDownloader(null, destination, null, null);
+
+    downloader.createDestinationDirectory();
+    // pass
+  }
+
+  @Test
+  public void testValidateAndCreateFiles_createDestinationAndPass() throws IOException {
+    Path destinationDir = tmp.getRoot().toPath().resolve("destinationDir");
+    Path destination = destinationDir.resolve("destinationFile");
+
+    StandardDownloader downloader = new StandardDownloader(null, destination, null, null);
+
+    Assert.assertFalse(Files.exists(destinationDir));
+    downloader.createDestinationDirectory();
+    Assert.assertTrue(Files.exists(destinationDir));
+  }
+
+  @Test
+  public void testValidateAndCreateFiles_targetDirectoryIsFile() throws IOException {
+    Path fileWithBadParent = tmp.newFile().toPath().resolve("xyz");
+
+    StandardDownloader downloader = new StandardDownloader(null, fileWithBadParent, null, null);
+    try {
+      downloader.createDestinationDirectory();
+      Assert.fail("NotDirectoryException expected but not thrown");
+    } catch (NotDirectoryException nde) {
+      // pass
+    }
+  }
+
+  @Test
+  public void testValidateAndCreateFiles_fileAlreadyExists() throws IOException {
+    Path fileThatExists = tmp.newFile().toPath();
+
+    StandardDownloader downloader = new StandardDownloader(null, fileThatExists, null, null);
+    try {
+      downloader.createDestinationDirectory();
+      Assert.fail("FileAlreadyExists expected but not thrown");
+    } catch (FileAlreadyExistsException faee) {
+      // pass
+    }
+  }
+
+  @Test
+  public void testDownloadURL_worksWithNullProgressListener()
+      throws IOException, InterruptedException {
+
+    Path destination = tmp.getRoot().toPath().resolve("destinationFile");
+    Path testSourceFile = createTestRemoteResource(1);
+    URL fakeRemoteResource = testSourceFile.toUri().toURL();
+
+    StandardDownloader downloader =
+        new StandardDownloader(fakeRemoteResource, destination, null, null);
+
+    downloader.downloadUrl();
```

Comment:
Wrap with a check for the return?

`Assert.assertEquals(destination, downloader.downloadUrl());`